### PR TITLE
Fixed ConfigurationOperation

### PR DESCRIPTION
**Pull request by clackner-gpa**
…s not exist

### CODE REVIEW

File: Source/Libraries/FaultData/DataOperations/ConfigurationOperation.cs [hunk 1]
```diff
@@ -461,11 +461,13 @@ string GetAggregatePrefix(double multiplier)
 
             Func<SourceIndex, bool> sourceIndexFilter = GetSourceIndexFilter(appDataType);
 
+            IEnumerable<object> tmp = sourceIndexMap
+                .SelectMany(item => Translate(item.SourceIndex), (item, SourceIndex) => new { item.Series, SourceIndex });
+
```
Comment (StephenCWills) [inline Source/Libraries/FaultData/DataOperations/ConfigurationOperation.cs:466]: Looks like code you added for debugging?

```suggestion
```

File: Source/Libraries/FaultData/DataOperations/ConfigurationOperation.cs [hunk 2]
```diff
@@ -513,14 +515,21 @@ int MatchCount(DataSeries dataSeries)
...
-            DataSeries calculatedSeries = sourceIndexes
-                .Where(sourceIndex => sourceIndex.ByName || sourceIndex.ChannelIndex >= 0)
-                .Select(sourceIndex => new { sourceIndex.Multiplier, DataSeries = FindDataSeries(sourceIndex) })
-                .Select(obj => obj.DataSeries.Multiply(obj.Multiplier))
-                .Aggregate((series1, series2) => series1.Add(series2));
+            IEnumerable<DataSeries> seriesParts = sourceIndexes
+                    .Where(sourceIndex => sourceIndex.ByName || sourceIndex.ChannelIndex >= 0)
+                    .Select(sourceIndex => new { sourceIndex.Multiplier, DataSeries = FindDataSeries(sourceIndex) })
+                    .Where(obj => !(obj.DataSeries is null))
+                    .Select(obj => obj.DataSeries.Multiply(obj.Multiplier));
```
Comment (StephenCWills) [inline Source/Libraries/FaultData/DataOperations/ConfigurationOperation.cs:522]:
```suggestion
            IEnumerable<DataSeries> seriesParts = sourceIndexes
                .Where(sourceIndex => sourceIndex.ByName || sourceIndex.ChannelIndex >= 0)
                .Select(sourceIndex => new { sourceIndex.Multiplier, DataSeries = FindDataSeries(sourceIndex) })
                .Where(obj => !(obj.DataSeries is null))
                .Select(obj => obj.DataSeries.Multiply(obj.Multiplier));
```

File: Source/Libraries/FaultData/DataOperations/ConfigurationOperation.cs [hunk 3]
```diff
@@ -513,14 +515,21 @@ int MatchCount(DataSeries dataSeries)
...
+
+            if (seriesParts.Count() == 0)
+                return;
 
+            DataSeries calculatedSeries = seriesParts
+                    .Aggregate((series1, series2) => series1.Add(series2));
```
Comment (StephenCWills) [inline Source/Libraries/FaultData/DataOperations/ConfigurationOperation.cs:528]:
```suggestion
                .Aggregate((series1, series2) => series1.Add(series2));
```

File: Source/Libraries/FaultData/DataOperations/ConfigurationOperation.cs [hunk 4]
```diff
@@ -513,14 +515,21 @@ int MatchCount(DataSeries dataSeries)
...
+              
             calculatedSeries.SeriesInfo = series;
             calculatedDataSeriesList.Add(calculatedSeries);
+
```
Comment (StephenCWills) [inline Source/Libraries/FaultData/DataOperations/ConfigurationOperation.cs:532]:
```suggestion
```

**Review by StephenCWills — CHANGES_REQUESTED** [review]

File: Source/Libraries/FaultData/DataOperations/ConfigurationOperation.cs (hunk 1 above)
Comment (clackner-gpa) [reply Source/Libraries/FaultData/DataOperations/ConfigurationOperation.cs]: Yes, I missed this one. Thank you for catching that

File: Source/Libraries/FaultData/DataOperations/ConfigurationOperation.cs [hunk 5]
```diff
@@ -513,12 +512,18 @@ int MatchCount(DataSeries dataSeries)
...
-            DataSeries calculatedSeries = sourceIndexes
+            IEnumerable<DataSeries> seriesParts = sourceIndexes
                 .Where(sourceIndex => sourceIndex.ByName || sourceIndex.ChannelIndex >= 0)
                 .Select(sourceIndex => new { sourceIndex.Multiplier, DataSeries = FindDataSeries(sourceIndex) })
-                .Select(obj => obj.DataSeries.Multiply(obj.Multiplier))
-                .Aggregate((series1, series2) => series1.Add(series2));
+                .Where(obj => !(obj.DataSeries is null))
+                .Select(obj => obj.DataSeries.Multiply(obj.Multiplier));
+
+            if (seriesParts.Count() == 0)
```
Comment (StephenCWills) [inline Source/Libraries/FaultData/DataOperations/ConfigurationOperation.cs:521]:
```suggestion
            if (!seriesParts.Any())
```

**Review by StephenCWills — APPROVED** [review]

**Pull request merged**